### PR TITLE
Add smoke test post deployment

### PR DESCRIPTION
**Pull request by RMcVelia**
## Description

Add smoke test to the build-and-deploy workflow post deployment

## Trello Card Link

https://trello.com/c/uYzoqzjx/430-irp-outstanding-deployment-tasks

## How to review

Check review app deploys and runs smoke test
run smoke test for any env e.g.
bin/smoke https://teacher-relocation-payment-pr-342.test.teacherservices.cloud current-sha

### CODE REVIEW

File: .github/actions/deploy/action.yml [hunk 1]
```diff
@@ -86,47 +86,11 @@ runs:
...
-#      with:
-#        environment: ${{ inputs.environment }}
-#        app-env: ${{ env.aks_app_environment }}
-#        pr-number: ${{ inputs.pr-number }}
-#        slack-webhook: ${{ inputs.slack-webhook }}
+    # - name: Run Smoke Tests for ${{ inputs.environment }}
+    #   uses: ./.github/workflows/smoke-test.yml
+    #   with:
+    #     environment: ${{ inputs.environment }}
+    #     url: ${{ steps.set_env_var.outputs.deploy_url }}
```
[the remark this answers is not in the document]
Comment (RMcVelia) [reply .github/actions/deploy/action.yml:93]: sorry, should be draft as I've been testing different things, and it's not in a reviewable state

[the remark this answers is not in the document]
Comment (fumimowdan) [reply .github/actions/deploy/action.yml:93]: :ok_hand:

File: bin/smoke [hunk 2]
```diff
@@ -1,8 +1,33 @@
-#!/usr/bin/env sh
-#
-# Application smoke test script
-#
+#!/usr/bin/env bash
 
-set -e
```
Comment (saliceti) [inline bin/smoke:6]: Can we keep this? It normally keeps the script more robust

File: bin/smoke [hunk 3]
```diff
@@ -1,8 +1,33 @@
...
+url=$1
```
Comment (saliceti) [inline bin/smoke:3]: Can we add a comment for usage: e.g. `./smoke <url> <commit id>`

File: .github/actions/deploy/action.yml [hunk 4]
```diff
@@ -86,47 +86,11 @@ runs:
...
-#      uses: ./.github/actions/smoke-test_v2/
-#      with:
-#        environment: ${{ inputs.environment }}
-#        app-env: ${{ env.aks_app_environment }}
-#        pr-number: ${{ inputs.pr-number }}
-#        slack-webhook: ${{ inputs.slack-webhook }}
+        current-commit-sha: ${{ inputs.sha }}
+        url: ${{ steps.set_env_var.outputs.deploy_url }}
 
 #    - name: Notify Slack channel on job failure
```
Comment (saliceti) [inline .github/actions/deploy/action.yml:95]: Can we enable Slack notifications?

File: bin/smoke (hunk 2 above)
Comment (RMcVelia) [reply bin/smoke:6]: done

File: bin/smoke (hunk 3 above)
Comment (RMcVelia) [reply bin/smoke:12]: done

File: .github/actions/deploy/action.yml (hunk 4 above)
Comment (RMcVelia) [reply .github/actions/deploy/action.yml:95]: added to backlog

**Review by saliceti — APPROVED** [review]

**Pull request merged**